pkg/ssh/ssh.go

```diff
@@ -0,0 +1,198 @@
+		return errors.Wrap(err, fmt.Sprintf("SSH public key file %q not found", filePath))
+	}
+
+	expandedPath := utils.ExpandPath(filePath)
+	fileContent, err := readFileContents(expandedPath)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("reading SSH public key file %q", filePath))
+	}
+
+	key := string(fileContent)
+
+	fingerprint, err := pki.ComputeAWSKeyFingerprint(key)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("computing fingerprint for key %q", filePath))
+	}
+	keyName := getKeyName(clusterName, ngName, fingerprint)
+
+	logger.Info("using SSH public key %q as %q ", expandedPath, keyName)
+
+	// Import SSH key in EC2
+	if err := importKey(keyName, fingerprint, &key, provider); err != nil {
+		return err
+	}
+	return nil
+}
+
+// LoadKeyByContent loads and imports an SSH public key into EC2 if it doesn't exist
+func LoadKeyByContent(key *string, clusterName, ngName string, provider api.ClusterProvider) error {
+	fingerprint, err := pki.ComputeAWSKeyFingerprint(*key)
+	if err != nil {
+		return errors.Wrap(err, fmt.Sprintf("computing fingerprint for key %q", *key))
+	}
+	keyName := getKeyName(clusterName, ngName, fingerprint)
+
+	logger.Info("using SSH public key %q ", *key)
+
+	// Import SSH key in EC2
+	if err := importKey(keyName, fingerprint, key, provider); err != nil {
+		return err
+	}
+	return nil
+}
+
+// DeleteKeys will delete the public SSH key, if it exists
+func DeleteKeys(clusterName string, provider api.ClusterProvider) {
+	existing, err := provider.EC2().DescribeKeyPairs(&ec2.DescribeKeyPairsInput{})
+	if err != nil {
+		logger.Debug("cannot describe keys: %v", err)
+		return
+	}
+	var matching []*string
+	prefix := getKeyName(clusterName, "", "")
+	logger.Debug("existing = %#v", existing)
+	for _, e := range existing.KeyPairs {
+		if !strings.HasPrefix(*e.KeyName, prefix) {
+			continue
+		}
+		nameParts := strings.Split(*e.KeyName, "-")
+		logger.Debug("existing key %q matches prefix", *e.KeyName)
+		if nameParts[len(nameParts)-1] == *e.KeyFingerprint {
+			logger.Debug("existing key %q matches fingerprint", *e.KeyName)
+			matching = append(matching, e.KeyName)
+		}
+	}
+	for i := range matching {
+		input := &ec2.DeleteKeyPairInput{
+			KeyName: matching[i],
+		}
+		logger.Debug("deleting key %q", *matching[i])
+		if _, err := provider.EC2().DeleteKeyPair(input); err != nil {
+			logger.Debug("key pair couldn't be deleted: %v", err)
+		}
+	}
+}
+
+// CheckKeyExistsInEC2 returns whether a public ssh key already exists in EC2 or error if it couldn't be checked
+func CheckKeyExistsInEC2(sshKeyName string, provider api.ClusterProvider) error {
+	existing, err := findKeyInEc2(sshKeyName, provider)
+	if err != nil {
+		return errors.Wrap(err, "checking existing key pair")
+	}
+
+	if existing == nil {
+		return fmt.Errorf("cannot find EC2 key pair %q", sshKeyName)
+	}
+
+	return nil
+}
+
+// FileExists returns true if there is a local file with that path
+func FileExists(filePath string) bool {
```

Comment:
I just remove function with the same name that was defined in `utils` (7203b1f1392668e462ef293f28a3203140772579) 🙄🤣...

Reply:
Perhaps we can create a new package in `pkg/utils/file` and move `ExpandPath` and this one there? We could also make sure that other instances of `if _, err := os.Stat(path); os.IsNotExist(err) {`  use the function, and it seems like  only `pkg/utils/kubeconfig` needs it basically.